app/components/Search/SearchContainer.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Body,
+  Icon,
+  Input,
+  Item,
+  Left,
+  ListItem,
+  Text,
+  View,
+} from 'native-base';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+import theme from '../../theme';
+import Autocomplete from 'native-base-autocomplete';
+import { useDispatch, useSelector } from 'react-redux';
+import * as stockActions from '../Stock/store/actions';
+import * as searchActions from './store/actions';
+import * as watchlistActions from '../Watchlist/store/actions';
+import { Actions as RouteActions } from 'react-native-router-flux';
+import { includes } from 'lodash';
+
+function SearchContainer() {
+  const [query, setQuery] = useState('');
+  const results = useSelector(({ search }) => search.data.results);
+  const symbols = useSelector(({ watchlist }) => watchlist.symbols);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(searchActions.getSymbols());
+  }, []);
+
+  useEffect(() => {
+    dispatch(searchActions.getResults(query));
+  }, [query]);
+
+  const handleSelectItem = item => {
+    dispatch(stockActions.setSymbol(item));
+    RouteActions.stockScreen();
+  };
+
+  const handleSearch = text => {
+    const q = text.trim();
+
+    setQuery(q);
+  };
+
+  const handleAddToWatchlist = symbol => {
+    return dispatch(watchlistActions.addWatchlistSymbol(symbol));
+  };
+
+  const handleWatchlistIconStyles = symbol => {
+    return includes(symbols, symbol) ? styles.negative : styles.positive;
+  };
+
+  const handleHideResults = results.length < 1 || query === '';
+
+  const renderWatchlistIcon = symbol => {
+    return includes(symbols, symbol)
+      ? 'md-remove-circle-outline'
+      : 'md-add-circle-outline';
+  };
+
+  const _renderInput = props => {
+    return (
+      <Item>
+        <Icon active name="search" />
+        <Input {...props} />
+      </Item>
+    );
+  };
+
+  const _renderListItem = result => {
+    const { symbol, companyName } = result;
+
+    return (
+      <ListItem key={symbol} onPress={() => handleSelectItem(symbol)}>
```

Comment:
Keys should be stable, predictable, and unique.you can't set like key something which is randomly generated.is not 100% that is unique everytime.If keys are not unique, can have an impact on performance.

Reply:
the `symbol` property is always unique.  No two results returned from search would ever have the same symbol.  The alternative is to use the array index but that would be very unstable as the array index would change any time `state.results` is updated - which I think would happen anytime the `query` updated.